### PR TITLE
doc: fix dead BUILD link

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong <loong.dai@intel.com>

### CODE REVIEW

File: docs/root/version_history/v1.16.0.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ Incompatible Behavior Changes
 -----------------------------
 *Changes that are expected to cause an incompatibility if applicable; deployment changes are likely required*
 
-* build: added visibility rules for upstream. If these cause visibility related breakage, see notes in :repo:`BUILD <v1.16:BUILD>`.
+* build: added visibility rules for upstream. If these cause visibility related breakage, see notes in :repo:`BUILD <https://github.com/envoyproxy/envoy/blob/release/v1.16/BUILD>`.
```
Comment (phlax) [inline docs/root/version_history/v1.16.0.rst:8]: im trying to rem how the `repo` tag works - i think it automatically expands the url, but maybe it needs to be hardcoded here, in which case im wondering about using the `repo` tag for this

**Comment by phlax** [comment]
rendered docs here https://storage.googleapis.com/envoy-pr/19415/docs/index.html

File: docs/root/version_history/v1.16.0.rst (hunk 1 above)
Comment (phlax) [reply docs/root/version_history/v1.16.0.rst:8]: @daixiang0 i think you also need to change `:repo:` -> `:ref:`
checking the rendered link here:

https://storage.googleapis.com/envoy-pr/2882a25/docs/version_history/v1.16.0.html

and it points to:

https://github.com/envoyproxy/envoy/blob/2882a2522a381f45139f942ec2144fd4edc8ee8f/https://github.com/envoyproxy/envoy/blob/release/v1.16/BUILD

**Review by phlax — CHANGES_REQUESTED** [review]
thanks for addressing this, if you update the link as described we can land

File: docs/root/version_history/v1.16.0.rst (hunk 1 above)
Comment (daixiang0) [reply docs/root/version_history/v1.16.0.rst:8]: Done.

@phlax thanks for review.

**Review by phlax — CHANGES_REQUESTED** [review]
almost there

im also wondering if there are any other `:repo:` links in the docs

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Pull request merged**